### DISCUSSION TITLE
씬 전환 후 다시 원래 씬으로 전환 시 캐릭터 동기화

### DISCUSSION THREAD

**Discussion by LeeYongSu1** [naverz/zepeto-studio-kor#559] [5. Q&A]
**World Ver: 1.1.5**

**App Ver: 3.10.4**

**구현하려고 하는 것**
멀티 플레이 씬에서 싱글 씬으로 전환 후 다시 멀티 플레이 씬으로 돌아오기

문제점
씬 전환 후 다시 원래 멀티 플레이 씬으로 돌아왔을 때 다른 플레이어의 움직임은 제 화면에 동기화가 되는데
본인 플레이어의 움직임이 다른 플레이어에게 동기화가 되지 않습니다.

씬 전환 시 캐릭터 생성 코드입니다. 
![화면 캡처 2022-03-31 100101](https://user-images.githubusercontent.com/80555875/160955353-c888a912-1e4d-4b36-8c2d-378272e85f48.png)
이 코드 외에는 샘플 멀티플레이 예제 코드를 그대로 쓰고 있습니다.

어떤 부분에 로직을 수정해야 다시 원래 씬으로 돌아왔을 때 다시 동기화가 될까요?

**Comment by Zepeto-TechEvangelist** [comment]
안녕하세요, 내 local 움직임을 서버에 보내는 코드는 

ClientStarter.ts의 SendTransform 부분입니다.


private SendTransform(transform: UnityEngine.Transform) {
        const data = new RoomData();

        const pos = new RoomData();
        pos.Add("x", transform.localPosition.x);
        pos.Add("y", transform.localPosition.y);
        pos.Add("z", transform.localPosition.z);
        data.Add("position", pos.GetObject());

        const rot = new RoomData();
        rot.Add("x", transform.localEulerAngles.x);
        rot.Add("y", transform.localEulerAngles.y);
        rot.Add("z", transform.localEulerAngles.z);
        data.Add("rotation", rot.GetObject());
        this.room.Send("onChangedTransform", data.GetObject());
    }

내 움직임 뿐 아니라 서버에 어떤 메세지를 보내는 방법은 this.room.Send(); 를 참고해서 구현하시면 좋을 것 같습니다.
`Room Message 가이드` : https://studio.zepeto.me/guides/multiplay-room-message
